Assignment 2.Rmd

```diff
@@ -87,13 +91,85 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
+```{r}
 
+EDGES.ST <- EDGE %>% group_by(to) %>%summarise(sum(count))
+EDGES.TE <- EDGE %>% group_by(from) %>% summarise(sum(count))
+names(EDGES.ST) <- c("id", "count")
+names(EDGES.TE) <- c("id", "count")
+VERTEX <- bind_rows(EDGES.ST, EDGES.TE)
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(g,layout=layout.fruchterman.reingold, vertex.size=VERTEX$count, edge.width=EDGE$count)
 
+```
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+library(dplyr)
+library(tidyr)
+```
+
+```{r}
+a <- read.csv("hudk4050-classes.csv", header=TRUE, na.strings = "")
+a <- a %>% filter(First.Name != "ZIMO") #Omit from the analysis because incomplete responses
+```
+
+### Unite first and last names into one column
+```{r}
+a <- a %>% unite(Name, First.Name, Last.Name, sep = ",")
+```
+
+### Gather courses
+```{r}
+a <- a %>% gather(ClassType, CourseName, Class.1, Class.2, Class.3,Class.4, Class.5, Class.6) %>% select(-c(ClassType))
+# Delete the sapcing between department and course number
+a$CourseName = gsub(" ", "", a$CourseName)
+a$Count <- 1
+# Omit  HUDK4050 because everyone is taking that class
+a  <- a %>% filter(CourseName != "HUDK4050")
+a <- na.omit(a)
+```
+### Spread
+```{r}
+a <- a %>% spread(CourseName, Count)
+
+rownames(a) = a$Name
+a$Name = NULL
+a = ifelse(is.na(a),0,1)
+```
+
+### Matrix
+```{r}
+b = as.matrix(a)
+c <- b %*% t(b)
+diag(c) =NA
+```
+
+### Graph
+```{r}
+d <- graph.adjacency(c, mode = "undirected")
+plot(d, layout = layout.fruchterman.reingold,vertex.size=10)
+
+```
+
+
+### Centrality
+```{r}
+degree(d)
+betweenness(d)
+```
+
+```{r}
+#
+```
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and degree. **Who is the most central person in the network?**
+
+### Central Person
+
+Since the graph for the class was really closed together, it was hard to tell who is the central person. This is the reason why I ran the code for degree and betweeness centrality. The degree centrality gives a score based on the number of links held by each node while the betweeness centrality measures the shortest path between other nodes. In other words, degree tells you who is the most popular while betweeness tells you who has the most influence around a system. This is why we use the score for degree to find out who is the most central person in the network. Based on the data above, Lintong Li is the most central person with the highest degree number of 48.
```

Comment:
Good explanation! One thing I would change though, betweeness measures the shortest paths that **pass through** a node.